### ISSUE TITLE
Connection to device succeeded but no datapoints found 

### ISSUE THREAD

**Issue by admiralalec** [rospogrigio/localtuya#1201]
## The problem
Device cannot be added to LocalTuya 

Error message: Connection to device succeeded but no datapoints found, please try again. Create a new issue and include debug logs if problem persists.

## Environment
- Localtuya version: v4.1.1
- Last working localtuya version (if known and relevant): never worked 
- Home Assistant Core version: 2022.12.5
- [] Are you using the Home Assistant Tuya Cloud component ? NO 
- [] Are you using the Tuya App in parallel ? YES, mobile app 

## Steps to reproduce
1. Add device through LocalTuya integration 
2. Used the cloud service to automatically get localkey etc 
(many other devices just work) 

## DP dump
```txt
{
  "result": {
    "active_time": 1662289536,
    "biz_type": 0,
    "category": "tdq",
    "create_time": 1662289536,
    "icon": "smart/solution/161001/4772fad51dca8c64_cover.png",
    "id": "",
    "ip": "",
    "lat": "",
    "local_key": "",
    "lon": "",
    "model": "2-433",
    "name": "Auto Gate Remote",
    "online": true,
    "owner_id": "",
    "product_id": "",
    "product_name": "2-433",
    "status": [
      {
        "code": "switch_1",
        "value": false
      },
      {
        "code": "switch_2",
        "value": false
      },
      {
        "code": "countdown_1",
        "value": 0
      },
      {
        "code": "countdown_2",
        "value": 0
      },
      {
        "code": "relay_status",
        "value": "0"
      },
      {
        "code": "random_time",
        "value": ""
      },
      {
        "code": "cycle_time",
        "value": ""
      },
      {
        "code": "switch_inching",
        "value": "AwACAQAC"
      }
    ],
    "sub": false,
    "time_zone": "+08:00",
    "uid": "",
    "update_time":,
    "uuid": ""
  },
  "success": true,
  "t":,
  "tid": ""
}
```

## Provide Home Assistant taceback/logs
```txt
Logger: custom_components.localtuya.pytuya
Source: custom_components/localtuya/pytuya/__init__.py:259
Integration: LocalTuya integration ([documentation](https://github.com/rospogrigio/localtuya/), [issues](https://github.com/rospogrigio/localtuya/issues))
First occurred: 23 December 2022 at 22:16:45 (6 occurrences)
Last logged: 17:16:08

[eb2...1si] Failed to get status:
[eb0...myq] Failed to get status:
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/asyncio/locks.py", line 390, in acquire
    await fut
asyncio.exceptions.CancelledError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/asyncio/tasks.py", line 456, in wait_for
    return fut.result()
asyncio.exceptions.CancelledError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 574, in detect_available_dps
    data = await self.status()
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 507, in status
    status = await self.exchange(STATUS)
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 486, in exchange
    msg = await self.dispatcher.wait_for(seqno)
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 259, in wait_for
    await asyncio.wait_for(self.listeners[seqno].acquire(), timeout=timeout)
  File "/usr/local/lib/python3.10/asyncio/tasks.py", line 458, in wait_for
    raise exceptions.TimeoutError() from exc
asyncio.exceptions.TimeoutError
```

**Comment by jim-downie** [comment]
I experienced this issue as well. Restarting Home Assistant after adding each device individually got round this problem (there may be variables not being reset after adding a device and restarting HA gets round this issue).

**Comment by admiralalec** [comment]
Thanks, but none of that works. 

Got the same issue with Tuya Moes GU10 Led Colour Bulbs (wifi).

**Comment by jim-downie** [comment]
Are all your existing Tuya devices online? Once I had added an incorrectly configured device through trying to force a DPS then my only option was to remove it, restart HA and re-add it back in (something went wrong in the initialisation of the extension during startup and every device I tried to add from then on would not discover the DPS's).

**Comment by admiralalec** [comment]
Understand, however all other devices work and can still be added.

**Comment by Border-of-the-void** [comment]
Can anyone suggest a solution to this problem?

**Comment by MoosJ80** [comment]
I have the same issue with a power meter.
I have tried to add it with a DP which i have found using this method:
https://www.zigbee2mqtt.io/advanced/support-new-devices/03_find_tuya_data_points.html#_9-find-your-data-point.

The entity stays unavailable.

**Comment by thedfordfox** [comment]
+1

Logger: homeassistant.helpers.frame
Source: helpers/frame.py:77
First occurred: 6:38:09 PM (1 occurrences)
Last logged: 6:38:09 PM

Detected integration that uses deprecated `async_get_registry` to access entity registry, use async_get instead. Please report issue to the custom integration author for localtuya using this method at custom_components/localtuya/config_flow.py, line 729: ent_reg = await er.async_get_registry(self.hass)

**Comment by sezlony** [comment]
+1

same error here as initial, I was hoping the missing v3.4 support was the problem, but now that there is, it still can't find datapoints 😢

**Comment by rospogrigio** [comment]
Did you try to use tinytuya with this device? Does it work?

**Comment by sezlony** [comment]
no way, I can't use linux for more than browsing...

**Comment by barrelltitor** [comment]
I have this same problem. I'm trying to move my HA to a docker container. I have two sockets added in my main HA via this component, however I cannot add them on my secondary installation.

I can add every other device, and my logs are identical to the others.

**Comment by ilmisu** [comment]
+1
Smart switch with no datapoints published.
log:

[bff...onm] Failed to get status:
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/asyncio/locks.py", line 390, in acquire
    await fut
asyncio.exceptions.CancelledError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/asyncio/tasks.py", line 456, in wait_for
    return fut.result()
asyncio.exceptions.CancelledError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 857, in detect_available_dps
    data = await self.status()
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 789, in status
    status = await self.exchange(DP_QUERY)
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 763, in exchange
    msg = await self.dispatcher.wait_for(seqno, payload.cmd)
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 450, in wait_for
    await asyncio.wait_for(self.listeners[seqno].acquire(), timeout=timeout)
  File "/usr/local/lib/python3.10/asyncio/tasks.py", line 458, in wait_for
    raise exceptions.TimeoutError() from exc
asyncio.exceptions.TimeoutError


any hint?
